src/utils.cc

```diff
@@ -113,7 +113,17 @@ std::string LowerPathIfInsensitive(const std::string &path) {
     c = tolower(c);
   return ret;
 #else
-  return path;
+  // use this opportunity to get full path
+  char *rpbuf = NULL;
+  std::string fullpath;
+  rpbuf = realpath(path.c_str(), NULL);
```

Comment:
I would say that since project is based on c++17 it worth to use `std::filesystem`

https://en.cppreference.com/w/cpp/filesystem/absolute